src/solver/mlir_common.cpp

```diff
@@ -41,6 +43,30 @@ std::string InsertGToLayout(const std::string& layout, char dim)
     return layout_with_g.insert(index, 1, 'G');
 }
 
+const char* DTypeName(miopenDataType_t ty) {
+    switch (ty) {
+    case miopenHalf:
+        return "fp16";
+    case miopenFloat:
+        return "fp32";
+    case miopenDouble:
+        return "fp64";
+    case miopenBFloat16:
+        return "bf16";
+    case miopenInt32:
+        return "i32";
+    case miopenInt8:
+        return "i8";
+    case miopenInt8x4:
+        return "i8x4";
```

Comment:
@whchung FYI, we are creating this 1:1 mapping string for data type representations. Right now, all except fp16, fp32 should be rejected by MLIR. To enable a new data type wouldn't require any change to MIOpen in the future.